### ISSUE TITLE
dnn::readNetFromTorch some layers not implemented 

### ISSUE THREAD

**Issue by bill2239** [opencv/opencv#10128]
##### System information (version)
<
- OpenCV => 3.3.1
- Operating System / Platform => Windows 64 Bit
- Compiler => Visual Studio 2015
>

##### Detailed description
I try to run use dnn:Net module to load torch model, and it get error in this line 
dnn::Net net = dnn::readNetFromTorch(modelfile);
error
OpenCV Error: The function/feature is not implemented (Unknown nn class "nn.BatchNormalization") in cv::dnn::experimental_dnn_v2::TorchImporter::readTorchObject,
##### Steps to reproduce
dnn::Net net = dnn::readNetFromTorch(modelfile);
I did convert the net to cpumode by net= net:float(), it seem like some layers are not implemented 
nn.Dropout also have the same error.

**Comment by dkurt** [comment]
@bill2239, please provide a minimal working example: a whole model definition or a single batch normalization layer test case.

**Comment by bill2239** [comment]
okay i see
model definition
local function ConvBNReLU(nInputPlane, nOutputPlane)
      vgg:add(backend.SpatialConvolution(nInputPlane, nOutputPlane, 3,3, 1,1, 1,1))
      vgg:add(nn.SpatialBatchNormalization(nOutputPlane,1e-3))
      vgg:add(backend.ReLU(true))
      return vgg
    end
    ConvBNReLU(1,64)--:add(nn.Dropout(0.3,nil,true))
    ConvBNReLU(64,64)
    vgg:add(MaxPooling(2,2,2,2):ceil())

    ConvBNReLU(64,128)--:add(nn.Dropout(0.4,nil,true))
    ConvBNReLU(128,128)
    vgg:add(MaxPooling(2,2,2,2):ceil())

    ConvBNReLU(128,256)--:add(nn.Dropout(0.4,nil,true))
    ConvBNReLU(256,256)--:add(nn.Dropout(0.4,nil,true))
    ConvBNReLU(256,256)
    vgg:add(MaxPooling(2,2,2,2):ceil())

    ConvBNReLU(256,512)--:add(nn.Dropout(0.4,nil,true))
    ConvBNReLU(512,512)--:add(nn.Dropout(0.4,nil,true))
    ConvBNReLU(512,512)
    vgg:add(MaxPooling(2,2,2,2):ceil())

    -- In the last block of convolutions the inputs are smaller than
    -- the kernels and cudnn doesn't handle that, have to use cunn
    backend = nn
    ConvBNReLU(512,512)--:add(nn.Dropout(0.4,nil,true))
    ConvBNReLU(512,512)--:add(nn.Dropout(0.4,nil,true))
    ConvBNReLU(512,512)
    vgg:add(MaxPooling(2,2,2,2):ceil())
    vgg:add(nn.View(512*2*7))

    classifier = nn.Sequential()
    --classifier:add(nn.Dropout(0.5,nil,true))
    classifier:add(nn.Linear(512*2*7,512))
    classifier:add(nn.BatchNormalization(512))
    classifier:add(nn.ReLU(true))
    --classifier:add(nn.Dropout(0.5,nil,true))
    classifier:add(nn.Linear(512,k*c))
    vgg:add(classifier)
    vgg:add(nn.Reshape(k,c))

**Comment by alalek** [comment]
@bill2239 Could you check this patch with your model: #10221 ?

**Comment by vermashresth** [comment]
@dkurt  Are there any other essential layers which exist in other deep learning libraries but aren't yet implemented in opencv dnn? I would like to try implementing them.

**Comment by dkurt** [comment]
@vermashresth, I'd be great! But I think it's pointless to implement just a separate layers. Try to find a whole interesting network that don't import by OpenCV. Or you may find a model that works differently comparing with origin framework. Of course you may take a look on existing layers and optimize their computations.
There is some issue about Keras library, https://github.com/opencv/opencv/issues/9894. It has a bit strange solution about relu6 layer.
Also we need to find a way to import Torch models made by `nngraph`: https://github.com/opencv/opencv/issues/9502.

**Comment by markmp** [comment]
I'm having exact same issue with a Keras trained mobilenet trying to get into OpenCV DNN. Was a workaround ever found for the relu6 layer with Minimum and Maximum?

**Comment by wolflion521** [comment]
@dkurt , Hi I wrote a custom layer in python, but get"Failed to create an instance of custom layer in function 'pycvLayer'". I just want to use the custom layer for tensorflow 'SplitV' op.
`class SplitVLayer(object):
    def __init__(self,params,blobs):
        self.num_or_size_splits = int(params["TLen"])
        self.axis = int(params["num_split"])
    def getmemoryShapes(self,inputs):
        inputShape = inputs[0]
        outshape = []
        for split in self.num_or_size_splits:
            outshape_temp = inputShape
            outshape_temp[self.axis] = split
            outshape.append(outshape_temp)
        return outshape
    def forward(self,inputs):
        return np.split(inputs[0],self.num_or_size_splits,self.axis)`

and I didn't forget using 
`cv2.dnn_registerLayer('SplitV',SplitVLayer)`
Can you tell me whether I am writing something wrong in the class? Thanks a lot.
